File: OpenPython/OutputFile.py

#
# This class provides an object model for an output text file.
#
# Author::    	Darrell O. Ricke, Ph.D.  (mailto: ricke@opengenes.com)
# Copyright:: 	Copyright (c) 2000 Darrell O. Ricke, Ph.D., Paragon Software
# License::   	GNU GPL license  (http://www.gnu.org/licenses/gpl.html)
# Contact::   	Paragon Software, 1314 Viking Blvd., Cedar, MN 55011
#
#             	This program is free software; you can redistribute it and/or modify
#             	it under the terms of the GNU General Public License as published by
#             	the Free Software Foundation; either version 2 of the License, or
#             	(at your option) any later version.
#         
#             	This program is distributed in the hope that it will be useful,
#             	but WITHOUT ANY WARRANTY; without even the implied warranty of
#             	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#             	GNU General Public License for more details.
#         
#             	You should have received a copy of the GNU General Public License
#             	along with this program; if not, write to the Free Software
#             	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
#
###############################################################################
class OutputFile:

  #
  # Initialization routine.
  #
  def initialize ( self ):
    self.file_name = ""			# File name
    self.file = None			# Output file
    self.lines = 0			# Lines written


  #############################################################################
  #
  # Constructor.
  #
  def __init__ ( self, name = "" ):
    self.initialize ()			# initialize
    self.setFileName ( name )
    self.openFile ()


  #############################################################################
  #
  # This method opens the output text file.
  #
  def openFile ( self ):
    if ( self.file_name == "" ):
      print("No file name has been specified.")
      return
    try:
      self.file = open ( self.file_name, "w" )	# open the file for writing
    except:
      print("Open failure on file named ", self.file_name)


  #############################################################################
  # 
  # This method closes the output text file.
  #
  def closeFile ( self ):
    self.file.close ()			# close the input file


  #############################################################################
  # 
  # This method deletes the output text file.
  #
  def deleteFile ( self ):
    print("The delete file function has not been implemented yet.")


  #############################################################################
  # 
  # This method returns the name of the output text file.
  #
  def getFileName ( self ):
    return self.file_name


  #############################################################################
  # 
  # This method sets the name of the output text file.
  #
  def setFileName ( self, name ):
    if ( name == "" ):  return
    self.file_name = str ( name )


  #############################################################################
  # 
  # This method writes to the output text file.
  #
  def write ( self, text ):
    self.file.write ( text )
    self.lines = self.lines + 1


###############################################################################
# 
# Testing module.
#
if ( __name__ == "__main__" ):
  test = OutputFile ( "test.data" )
  # for i in range ( 10 ):
  #   test.write ( "Line " + str ( i ) + "\n" )
  test.write( ">Seq1 This is a FASTA sequence file.\n" )
  test.write( "ACGTACGTACGT\n" )
  test.write( ">Seq2 This second sequence in the file.\n" )
  test.write( "AAAACCCCGGGGTTTT\n" )
  test.closeFile ()

File: OpenRuby/output_file.rb

# This class provides an object model for an output text file.
#
# Author::      Darrell O. Ricke, Ph.D.  (mailto: d_ricke@yahoo.com)
# Copyright::   Copyright (C) 2000 Darrell O. Ricke, Ph.D., Paragon Software
# License::     GNU GPL license:  http://www.gnu.org/licenses/gpl.html
# Contact::     Paragon Software, 1314 Viking Blvd., Cedar, MN 55011
#
#    This program is free software: you can redistribute it and/or modify
#    it under the terms of the GNU General Public License as published by
#    the Free Software Foundation, either version 2 of the License, or
#    (at your option) any later version.
#
#    This program is distributed in the hope that it will be useful,
#    but WITHOUT ANY WARRANTY; without even the implied warranty of
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#    GNU General Public License for more details.
#
#    You should have received a copy of the GNU General Public License
#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
         
###############################################################################
class OutputFile

# lines - the number of lines written to the output text file.
attr_accessor  :lines
# file_name - the name of the output text file.
attr_accessor :file_name


###############################################################################
# Create an output object for the named file.
def initialize( name )
  @lines = 0				  # lines written
  close_file()				# close the file
  @file = nil
  @file_name = name 	# set the file name
end  # method initialize


###############################################################################
# Open the output text file.
def open_file()
  @file = File.new( @file_name, "w" )
end  # method open_file


###############################################################################
# Close the output text file.
def close_file()
  @file.close unless @file.nil?	# close file
end  # method close_file


###############################################################################
# Delete the file.
def delete_file()
  close_file()				        # close the file if open
  File.delete( @file_name )		# delete the file by name
end  # method delete_file 


###############################################################################
# Write a text string to the output text file.
def write( text )
  begin
    @file.print( text ) unless @file.nil?
    @lines += 1
  rescue
    print "Write error"
  end  # begin
end  # method write 


###############################################################################
# Write a binary data to the output text file.
def write_binary(data)
  begin
    @file.binmode
    @file.write(data) unless @file.nil?
  rescue
    print "write_binary error"
  end
end  # method write_binary

end  # class OutputFile


###############################################################################
# Testing module.
def test_output_file
  file = OutputFile.new( "test.data" )
  file.open_file()
  file.write( ">Seq1 This is a FASTA sequence file.\n" )
  file.write( "ACGTACGTACGT\n" )
  file.write( ">Seq2 This second sequence in the file.\n" )
  file.write( "AAAACCCCGGGGTTTT\n" )
  file.close_file()
  print "The number of lines written is ", file.lines, "\n"
end  # method test_output_file

# test_output_file() 

